bftengine/src/preprocessor/messages/ClientBatchRequestMsg.cpp

```diff
@@ -60,6 +60,36 @@ void ClientBatchRequestMsg::validate(const ReplicasInfo& repInfo) const {
     LOG_ERROR(logger(), "Message sender is invalid" << KVLOG(senderId()));
     throw std::runtime_error(__PRETTY_FUNCTION__);
   }
+  if (!checkElements()) {
+    LOG_ERROR(logger(), "One or more ClientMsg in the list is invalid");
+    throw std::runtime_error(__PRETTY_FUNCTION__);
+  }
+}
+bool ClientBatchRequestMsg::checkElements() const {
+  const auto totalMsgSize = size();
+  const auto& numOfMessagesInBatch = msgBody()->numOfMessagesInBatch;
```

Comment:
I think we should consider a macro MAX_NUM_ELEMENT_IN_BATCH. You can set it for now to 200 for example, but maybe reserch a little more (speak with some other people) on the exact max value. If you are checking for validity, every data can be corrupted. Therfor, also the number of messages. If this number is 0 or too big, we should also drop.

Reply:
Added check for MAX_BATCH_SIZE